fixed-hash/src/hash.rs

```diff
@@ -485,9 +535,110 @@ macro_rules! construct_fixed_hash {
 				hash
 			}
 		}
+	}
+}
 
-		#[cfg(feature = "rustc-hex-support")]
-		impl $crate::core::str::FromStr for $name {
+// Implementation for disabled libc crate support.
+// 
+// # Note
+//
+// Feature guarded macro definitions instead of feature guarded impl blocks
+// to work around the problems of introducing `libc` crate feature in
+// a user crate.
+#[cfg(not(all(feature = "libc", not(target_os = "unknown"))))]
+#[macro_export]
+#[doc(hidden)]
+macro_rules! impl_libc_for_fixed_hash {
+	( $name:ident ) => {
+		impl $crate::core_::cmp::PartialEq for $name {
+			#[inline]
+			fn eq(&self, other: &Self) -> bool {
+				self.as_bytes() == other.as_bytes()
+			}
+		}
+
+		impl $crate::core_::cmp::Ord for $name {
+			#[inline]
+			fn cmp(&self, other: &Self) -> $crate::core_::cmp::Ordering {
+				self.as_bytes().cmp(other.as_bytes())
+			}
+		}
+	}
+}
+
+// Implementation for enabled libc crate support.
+// 
+// # Note
+//
+// Feature guarded macro definitions instead of feature guarded impl blocks
+// to work around the problems of introducing `libc` crate feature in
+// a user crate.
+#[cfg(all(feature = "libc", not(target_os = "unknown")))]
+#[macro_export]
+#[doc(hidden)]
+macro_rules! impl_libc_for_fixed_hash {
```

Comment:
I still have not been able to understand what advantage libc support actually brings. Is it more performant to impl `PartialEq`/`Ord` like this?

Reply:
I will just merge this now and we can resolve this issue in a later PR, is that okay?

Reply:
Sure, just open an issue for it please?

Reply:
Personally, I don't want to block merging this PR because of this.

Reply:
will do

Reply:
done: https://github.com/paritytech/parity-common/issues/79

Reply:
merging now